### ISSUE TITLE
[Question] Accessibility Snapshot Fails to Capture Content in Web Office Applications

### ISSUE THREAD

**Issue by HarishwarTG** [microsoft/playwright-mcp#562]
I've noticed that the Playwright MCP's accessibility snapshot feature is unable to capture content data within web-based Office applications, such as Word and Excel. This limitation affects the ability to accurately assess and test accessibility in these environments.

Please confirm if this is an expected behavior or a potential bug? Any insights or workarounds would be greatly appreciated.

**Comment by Skn0tt** [comment]
This is definitely a bug. Does it occur for any Word file for you, or only for some? I can take a look at reproducing it. Might be related to https://github.com/microsoft/playwright-mcp/issues/535.

**Comment by HarishwarTG** [comment]
Yes, I have tested with few MS web Word and Excel file. It was not able to capture the content inside the file. 

Repro by creating a file with random data
https://excel.cloud.microsoft/ 

Thanks for swift response👍

**Comment by Skn0tt** [comment]
Excel is using `canvas` elements to draw the spreadsheets. They don't have any accesibility info in them, so there's nothing for us to show:

<img width="1766" alt="Image" src="https://github.com/user-attachments/assets/1c366199-0385-49b1-9a66-8135d4a46777" />

I'm a little surprised at this lack of a11y in the core part of the app, given that Microsoft apps are usually pretty good at that. Maybe there's a hidden option to enable or something?

**Comment by HarishwarTG** [comment]
Thanks!

I remember It was working last week, now I'm trying to figure out if the issue is due to a change on Microsoft's end or something that changed in Playwright🤔

**Comment by Skn0tt** [comment]
I'm able to navigate the app properly using MacOS VoiceOver, so it's not an a11y problem with the site itself. The ARIA tree in devtools still looks wrong. I'll investigate.

**Comment by Skn0tt** [comment]
Here's a video that demonstrates how Excel implements a11y. When they detect that a screenreader is enabled, they add a "readout" aria element to DOM. When a cell is selected, its value is written into the element, and the screenreader reads it out.  You can see that element change on the middle right side of the screen, as i'm cycling through the cells:

https://github.com/user-attachments/assets/7c55c1da-0908-49a2-b18b-58641ae7a0a1

I'm not sure if this works for human usage of the page, but it's definitely not compatible with how Playwright MCP perceives the page. As a workaround, use `--vision` mode when interacting with Office apps, or use a purpose-built Office Agent / MCP server.
